flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/TestingFlinkKubeClient.java

```diff
@@ -149,9 +149,10 @@ public List<KubernetesPod> getPodsWithLabels(Map<String, String> labels) {
     }
 
     @Override
-    public KubernetesWatch watchPodsAndDoCallback(
+    public CompletableFuture<KubernetesWatch> watchPodsAndDoCallback(
             Map<String, String> labels, WatchCallbackHandler<KubernetesPod> podCallbackHandler) {
-        return watchPodsAndDoCallbackFunction.apply(labels, podCallbackHandler);
+        return CompletableFuture.supplyAsync(
+                () -> watchPodsAndDoCallbackFunction.apply(labels, podCallbackHandler));
```

Comment:
I think we should change the return value type of `watchPodsAndDoCallbackFunction` to align with the method signature.